### PR TITLE
fix(nice-grpc): ensure header is sent with first response even if `context.sendHeader()` is not called

### PR DESCRIPTION
**Pull request by aikoven**

### CODE REVIEW

File: packages/nice-grpc/src/server/handleUnaryCall.ts [hunk 1]
```diff
@@ -57,30 +57,34 @@ export function createUnaryMethodHandler<Request, Response>(
...
-            continue;
-          }
+              continue;
+            }
 
-          return result.value;
+            return result.value;
+          }
+        } finally {
+          context.sendHeader();
```
Comment (rostik404) [inline packages/nice-grpc/src/server/handleUnaryCall.ts:87]: Hello!
Could you please tell me what the reason is for sending the header here? With these changes, grpc-retries do not work and the client receives empty metadata in response. Without this line, retries work properly, and metadata is not empty.
I used test client from grpc examples https://github.com/grpc/grpc-java/tree/master/examples/src/main/java/io/grpc/examples/retrying

Comment (rostik404) [reply packages/nice-grpc/src/server/handleUnaryCall.ts:87]: @aikoven

Comment (aikoven) [reply packages/nice-grpc/src/server/handleUnaryCall.ts:87]: Hey @rostik404!

Without this line, the header would not be sent unless the user calls `context.sendHeader()` explicitly. If the user does not call it, we send the header just before sending the response.

You can open an issue with reproduction steps — I'll take a look!

Comment (rostik404) [reply packages/nice-grpc/src/server/handleUnaryCall.ts:87]: Thank you, I created an issue https://github.com/deeplay-io/nice-grpc/issues/722

**Pull request merged**